error[E0277]: the trait bound `i32: rocket::http::uri::FromUriParam<&str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:25:23
   |
25 |     uri!(simple: id = "hi"); //~ ERROR i32: rocket::http::uri::FromUriParam<&str>
   |                       ^^^^ the trait `rocket::http::uri::FromUriParam<&str>` is not implemented for `i32`
   |
   = note: required by `rocket::http::uri::FromUriParam::from_uri_param`

error[E0277]: the trait bound `i32: rocket::http::uri::FromUriParam<&str>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:26:18
   |
26 |     uri!(simple: "hello"); //~ ERROR i32: rocket::http::uri::FromUriParam<&str>
   |                  ^^^^^^^ the trait `rocket::http::uri::FromUriParam<&str>` is not implemented for `i32`
   |
   = note: required by `rocket::http::uri::FromUriParam::from_uri_param`

error[E0277]: the trait bound `i32: rocket::http::uri::FromUriParam<i64>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:27:23
   |
27 |     uri!(simple: id = 239239i64); //~ ERROR i32: rocket::http::uri::FromUriParam<i64>
   |                       ^^^^^^^^^ the trait `rocket::http::uri::FromUriParam<i64>` is not implemented for `i32`
   |
   = note: required by `rocket::http::uri::FromUriParam::from_uri_param`

error[E0277]: the trait bound `S: rocket::http::uri::FromUriParam<_>` is not satisfied
  --> $DIR/typed-uri-bad-type.rs:28:31
   |
28 |     uri!(not_uri_display: 10, S); //~ ERROR S: rocket::http::uri::FromUriParam<_>
   |                               ^ the trait `rocket::http::uri::FromUriParam<_>` is not implemented for `S`

error: aborting due to 4 previous errors

For more information about this error, try `rustc --explain E0277`.
